subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/serialization/codecs/TaskReferenceCodec.kt

```diff
@@ -28,28 +30,44 @@ internal
 object TaskReferenceCodec : Codec<Task> {
 
     override suspend fun WriteContext.encode(value: Task) {
-        if (value === isolate.owner.delegate) {
+        val owner = isolate.owner
+        val delegate = owner.delegate
+
+        if (value === delegate) {
             writeBoolean(true)
         } else {
-            logUnsupported(
-                "serialize",
-                Task::class,
-                value.javaClass,
-                documentationSection = RequirementsTaskAccess
-            )
             writeBoolean(false)
+            val isTaskReferencesAllowed = owner is IsolateOwner.OwnerTask && owner.allowTaskReferences
```

Comment:
🔧 Extract complex expression to method so the `if` predicate below reads `if (isTaskReferenceAllowedBy(owner, task)) `